### PR TITLE
wip-mariocao-min-difficulty-dr: draft submission

### PR DESCRIPTION
**Pull request by mariocao**
> This proposal introduces a lower bound on data request mining difficulty in case that the Active Reputation Set (ARS) goes to abnormally low sizes to avoid 2 undesirable scenarios:
>
> 1. if ARS size is slightly bigger than the data request replication factor, mining is mostly taken over by a few identities therein, and
> 2. if ARS is empty or almost empty, most identities (even those out of the ARS) are eligible to mine data requests as their probability will tend to 100%.

This solves #56

### CODE REVIEW

File: wip-mariocao-min-difficulty-dr.md
Comment (tmpolaczyk) [inline wip-mariocao-min-difficulty-dr.md:169]: Different? Other?

```suggestion
- Non-implementers MAY consolidate different blocks and superblocks than implementers.
```

File: wip-mariocao-min-difficulty-dr.md
Comment (aesedepece) [inline wip-mariocao-min-difficulty-dr.md:110]:
```suggestion
5. The remainder of the `offset_reputation` integer division is assigned (in order) to the sorted identities until depleted (see _Step 5: remaining_ in Fig.2).
```

File: wip-mariocao-min-difficulty-dr.md
Comment (aesedepece) [inline wip-mariocao-min-difficulty-dr.md:130]:
```suggestion
In case of abnormally low ARS sizes, it is assumed that there is a number of non-reputed identities in the order of thousands computing their eligibility to be selected to solve the data request. This assumption is in line with the minimum difficulty set in [WIP-0012] and previous network conditions documented in [WIP-0009].
```

File: wip-mariocao-min-difficulty-dr.md
Comment (aesedepece) [inline wip-mariocao-min-difficulty-dr.md:144]:
```suggestion
Analogously to [WIP-0012], for such lower bound to be effective, its value needs to be strictly greater than `replication_factor`, as otherwise, ARS members will have an eligibility equal or greater than 100%.
```

File: wip-mariocao-min-difficulty-dr.md
Comment (aesedepece) [inline wip-mariocao-min-difficulty-dr.md:146]: I think we could omit this detail that gives some color on WIP-0012 but doesn't contribute much to this specification.

File: wip-mariocao-min-difficulty-dr.md
Comment (aesedepece) [inline wip-mariocao-min-difficulty-dr.md:161]:
```suggestion
- Blocks and transactions that were considered valid by former versions of the protocol MUST continue to be considered valid.
```

**Review by aesedepece — CHANGES_REQUESTED** [review]

**Comment by aesedepece** [comment]
As provided for by [WIP-0001](https://github.com/witnet/WIPs/blob/master/wip-0001.md), given that all the formal requirements are met, upon the [author's request](https://github.com/witnet/WIPs/pull/59#issuecomment-860591618), I'm:
- Assigning this draft the WIP number 0016
- Approving this PR for public listing on the WIP repository
- Call the author to address the comments provided before merging the draft

@mariocao you the green light to rename from `wip-mariocao-min-difficulty-dr` to `wip-0016`.

**Comment by aesedepece** [comment]
Once this draft is published, when elaborating an adoption plan, let's take into account that this WIP is a great candidate for activation through [WIP-0014 (TAPI)](https://github.com/witnet/WIPs/blob/master/wip-0014.md).

File: wip-mariocao-min-difficulty-dr.md
Comment (mariocao) [reply wip-mariocao-min-difficulty-dr.md:74]: > What's the subtext behind the quotation marks?

First, it was to remind me to review that adjective.
I wanted to avoid using "fairer distribution", which can be considered a subjective qualifier.

Anyhow, I forgot to remove the double quotes.

**Review by aesedepece — APPROVED** [review]
Thanks for addressing all the comments!

:1st_place_medal: I'm approving this PR for public listing on the WIP repository

@mariocao you have the green light to rename from `wip-mariocao-min-difficulty` to `wip-0016` and merge this PR.

**Pull request merged**
